data-prepper-expression/src/test/java/org/opensearch/dataprepper/expression/UnaryNumericOperatorTest.java

```diff
@@ -0,0 +1,129 @@
+    private static final int SYMBOL = RANDOM.nextInt();
+    private static final String DISPLAY_NAME = UUID.randomUUID().toString();
+
+    @Mock
+    private Map<Class<? extends Number>, Function<Number, ? extends Number>> strategy;
+
+    @Mock
+    private RuleContext ctx;
+
+    private UnaryNumericOperator unaryNumericOperator;
+
+    @BeforeEach
+    void beforeEach() {
+        unaryNumericOperator = new UnaryNumericOperator(SYMBOL, DISPLAY_NAME, strategy);
+    }
+
+    @Test
+    void testGivenUnaryOperatorExpressionThenShouldEvaluateReturnTrue() {
+        doReturn(DataPrepperExpressionParser.RULE_unaryOperatorExpression)
+                .when(ctx)
+                .getRuleIndex();
+        assertThat(unaryNumericOperator.shouldEvaluate(ctx), is(true));
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {
+            DataPrepperExpressionParser.RULE_expression,
+            DataPrepperExpressionParser.RULE_conditionalExpression,
+            DataPrepperExpressionParser.RULE_conditionalOperator,
+            DataPrepperExpressionParser.RULE_equalityOperatorExpression,
+            DataPrepperExpressionParser.RULE_equalityOperator,
+            DataPrepperExpressionParser.RULE_regexOperatorExpression,
+            DataPrepperExpressionParser.RULE_regexEqualityOperator,
+            DataPrepperExpressionParser.RULE_relationalOperatorExpression,
+            DataPrepperExpressionParser.RULE_relationalOperator,
+            DataPrepperExpressionParser.RULE_setOperatorExpression,
+            DataPrepperExpressionParser.RULE_setOperator,
+            DataPrepperExpressionParser.RULE_parenthesesExpression,
+            DataPrepperExpressionParser.RULE_regexPattern,
+            DataPrepperExpressionParser.RULE_setInitializer,
+            DataPrepperExpressionParser.RULE_unaryNotOperatorExpression,
+            DataPrepperExpressionParser.RULE_unaryOperator,
+            DataPrepperExpressionParser.RULE_primary,
+            DataPrepperExpressionParser.RULE_jsonPointer,
+            DataPrepperExpressionParser.RULE_variableIdentifier,
+            DataPrepperExpressionParser.RULE_variableName,
+            DataPrepperExpressionParser.RULE_literal
+
+    })
+    void testGivenNotUnaryOperatorExpressionThenShouldEvaluateReturnFalse(final Integer ruleIndex) {
+        doReturn(ruleIndex)
+                .when(ctx)
+                .getRuleIndex();
+        assertThat(unaryNumericOperator.shouldEvaluate(ctx), is(false));
+    }
+
+    @Test
+    void testGetSymbol() {
+        assertThat(unaryNumericOperator.getSymbol(), is(SYMBOL));
+    }
+
+    @ParameterizedTest
+    @MethodSource("argProvider")
+    void testGivenSingleArgThenEvaluateCallsStrategyFunction(final Number arg, final Class<? extends Number> clazz) {
+        final AtomicBoolean isStrategyFunctionCalled = new AtomicBoolean(false);
+        final Number expected = RANDOM.nextInt();
+        final Function<Number, ? extends Number> strategyFunction = param -> {
```

Comment:
You could use Mockito here. `mock(Function.class)`, `when(function.apply(any)).thenReturn(expected)`, and `verify(function).apply(any())`

I think it will be clearer.